src/shaders_140/shapecorners.frag

```diff
@@ -24,7 +24,7 @@ vec4 shadowCorner(float distance_from_center) {
     if(percent < 0)
         return vec4(0,0,0,0);
     else
-        return vec4(shadowColor.rgb, shadowColor.a * percent);
+        return vec4(shadowColor.rgb * (shadowColor.a / 1) * percent, shadowColor.a * percent);
```

Comment:
Can you test if this is equivalent to `return shadowColor * percent;` ?

Reply:
It seems the shader crashed, and every window turned black. Also, I don't think they are the same since `shadowColor.rgb` also need to multiply to `shadowColor.a` as equivalent to `tex.rgb *= tex.a;`, or the shadow will turn white. But I just realize the `/ 1` is redundant.

Reply:
Oh okay. Can you also make the same change in `shaders_110`?

Reply:
I don't see `tex.rgb *= tex.a` in `shaders_110`. Maybe it is not needed for `shaders_110`? I am not sure when the `shaders_110` was used.

Reply:
OMG, you are right! I must have missed that. I just added the modulation adjustment in `shaders_110`.  But I left the `shadowCorner` function for you so it would be your contribution.

I personally prefer to keep both versions of shader similar to each other and keep their syntax differences only.

Reply:
Thanks. I made the same changes for `shaders_110`.